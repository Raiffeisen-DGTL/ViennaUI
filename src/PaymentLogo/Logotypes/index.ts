export { MasterCard } from './MasterCard';
export { Mir } from './Mir';
export { Visa } from './Visa';
