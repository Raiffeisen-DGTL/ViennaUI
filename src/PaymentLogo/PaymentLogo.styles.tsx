import styled, { css } from 'styled-components';
import { paymentlogo } from 'vienna.ui-theme';
import { getPresets } from '../Utils/styling';

const presets = getPresets(
    paymentlogo,
    'paymentlogo'
)({
    base: null,
    size: '$size',
    design: '$design',
    custom: null,
});

export interface PropsBox {
    $design?: 'wildsand' | 'whitebox' | 'ghost';
    $size?: 'xs' | 's' | 'm' | 'l' | 'xl';
    $clickable?: boolean;
}
export const Box = styled.div<PropsBox>`
    display: inline-flex;
    justify-content: center;
    align-items: center;
    box-sizing: border-box;
    white-space: nowrap;
    user-select: none;
    appearance: none;
    outline: 0;
    position: relative;

    ${presets.base}
    ${presets.size}
    ${presets.design}

    ${({ $clickable }) =>
        $clickable &&
        css`
            cursor: pointer;
        `}

    ${presets.custom}
`;
