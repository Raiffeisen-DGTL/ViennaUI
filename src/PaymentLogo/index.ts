export { PaymentLogo, type PaymentLogoProps } from './PaymentLogo';
