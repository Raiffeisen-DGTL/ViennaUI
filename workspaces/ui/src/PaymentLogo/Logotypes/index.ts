export * from './MasterCard';
export * from './Mir';
export * from './Visa';
