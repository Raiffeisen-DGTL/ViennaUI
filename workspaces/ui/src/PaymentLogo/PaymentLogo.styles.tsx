import styled, { css } from 'styled-components';
import { getPresets } from 'vienna.ui-primitives';

const presets = getPresets('paymentlogo', {
    base: null,
    size: 'size',
    design: 'design',
    custom: null,
});

export const Box = styled.div<any>`
    display: inline-flex;
    justify-content: center;
    align-items: center;
    box-sizing: border-box;
    white-space: nowrap;
    user-select: none;
    appearance: none;
    outline: 0;
    position: relative;

    ${presets.base}
    ${presets.size}
    ${presets.design}

    ${(props) =>
        props.clickable &&
        css`
            cursor: pointer;
        `}

    ${presets.custom}
`;
