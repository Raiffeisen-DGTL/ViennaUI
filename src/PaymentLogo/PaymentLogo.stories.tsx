import React from 'react';
import { Story, Meta } from 'storybook';
import { PaymentLogo, PaymentLogoProps } from './PaymentLogo';

export default {
    title: 'Development/PaymentLogo',
    component: PaymentLogo,
} as Meta;

export const Overview: Story<PaymentLogoProps> = (args) => {
    return <PaymentLogo {...args} />;
};
