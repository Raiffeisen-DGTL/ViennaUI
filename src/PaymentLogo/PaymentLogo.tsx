import React, { forwardRef } from 'react';
import { MasterCard, Mir, Visa } from './Logotypes';
import { Box, PropsBox } from './PaymentLogo.styles';
import { BoxStyled } from '../Utils/styled';

export interface PaymentLogoProps extends BoxStyled<typeof Box, PropsBox> {
    design?: PropsBox['$design'];
    size?: PropsBox['$size'];
    logo?: 'mastercard' | 'mir' | 'visa';
}

const logotypes = {
    mastercard: MasterCard,
    mir: Mir,
    visa: Visa,
};

export const PaymentLogo = forwardRef<HTMLDivElement, PaymentLogoProps>((props, ref) => {
    const { children, size = 'm', logo = 'mastercard', design = 'wildsand', ...attrs } = props;

    const Logotype = logotypes[logo];

    return (
        <Box {...attrs} ref={ref} $size={size} $clickable={!!props.onClick} $design={design}>
            {children ?? <Logotype size={size} />}
        </Box>
    );
});

PaymentLogo.displayName = 'PaymentLogo';
