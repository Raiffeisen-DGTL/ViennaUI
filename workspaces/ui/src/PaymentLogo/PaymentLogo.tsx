import React, { HTMLAttributes } from 'react';
import { MasterCard, Mir, Visa } from './Logotypes';
import { Box } from './PaymentLogo.styles';

export interface PaymentLogoProps extends Omit<HTMLAttributes<HTMLDivElement>, 'size'> {
    children?: React.ReactNode;
    logo?: 'mastercard' | 'mir' | 'visa';
    design?: 'wildsand' | 'whitebox' | 'ghost';
    size?: 'xs' | 's' | 'm' | 'l' | 'xl';
    clickable?: boolean;
    forwardedRef?: any;
}

export const PaymentLogo: React.FC<PaymentLogoProps> = (props): JSX.Element => {
    const { children, size, logo = 'mastercard', forwardedRef, ...attrs } = props;

    const clickable = props.onClick ?? props.clickable;
    const logotypes = { mastercard: MasterCard, mir: Mir, visa: Visa };

    return (
        <Box {...attrs} size={size} ref={forwardedRef} clickable={clickable}>
            {children ?? React.createElement(logotypes[logo], { size }, null)}
        </Box>
    );
};

PaymentLogo.defaultProps = {
    size: 'm',
    design: 'wildsand',
    logo: 'mastercard',
};

PaymentLogo.displayName = 'PaymentLogo';
export default PaymentLogo;
