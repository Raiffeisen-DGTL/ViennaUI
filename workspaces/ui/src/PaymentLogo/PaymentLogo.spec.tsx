import React from 'react';
import PaymentLogo, { PaymentLogoProps } from './PaymentLogo';

test('PaymentLogo', () => {
    const snap = snapshot.render(<PaymentLogo />);
    expect(snap).toMatchSnapshot();
});

test('PaymentLogo w/ children', () => {
    const snap = snapshot.render(<PaymentLogo>Logo</PaymentLogo>);
    expect(snap).toMatchSnapshot();
});

test('PaymentLogo w/ logo', () => {
    const logos: PaymentLogoProps['logo'][] = ['mastercard', 'mir', 'visa'];

    logos.forEach((logo) => {
        const snap = snapshot.render(<PaymentLogo logo={logo} />);
        expect(snap).toMatchSnapshot(logo);
    });
});

test('PaymentLogo w/ design', () => {
    const designs: PaymentLogoProps['design'][] = ['wildsand', 'whitebox', 'ghost'];

    designs.forEach((design) => {
        const snap = snapshot.render(<PaymentLogo design={design} />);
        expect(snap).toMatchSnapshot(design);
    });
});

test('PaymentLogo w/ size', () => {
    const sizes: PaymentLogoProps['size'][] = ['xs', 's', 'm', 'l', 'xl'];

    sizes.forEach((size) => {
        const snap = snapshot.render(<PaymentLogo size={size} />);
        expect(snap).toMatchSnapshot(size);
    });
});

test('PaymentLogo w/ clickable', () => {
    const snap = snapshot.render(<PaymentLogo clickable />);
    expect(snap).toMatchSnapshot();
});

test('PaymentLogo w/ any props', () => {
    const props = { className: 'className', id: 'alarm', title: 'title', onClick: () => 'PaymentLogo' };
    const component = <PaymentLogo {...props} />;

    const snap = snapshot.render(component);
    expect(snap).toMatchSnapshot();

    const wrapper = snapshot.shallow(component);
    for (const [key, value] of Object.entries(props)) {
        expect(wrapper.props()[key]).toEqual(value);
    }
});
