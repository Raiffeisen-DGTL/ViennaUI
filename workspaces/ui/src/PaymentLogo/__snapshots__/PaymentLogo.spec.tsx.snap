// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaymentLogo 1`] = `
<div
  class="PaymentLogostyles__Box-sc-1op762m-0 jmiTAj"
>
  <svg
    height="16"
    viewBox="0 0 28 16"
    width="28"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      clip-rule="evenodd"
      d="M9.713 1.997h9.313v11.98H9.713V1.996z"
      fill="#FF5F00"
      fill-rule="evenodd"
    />
    <path
      d="M13.97 13.963A8.58 8.58 0 018.383 16C3.752 16 0 12.418 0 8s3.753-8 8.382-8a8.58 8.58 0 015.588 2.037C12.256 3.502 11.176 5.631 11.176 8c0 2.37 1.08 4.498 2.794 5.963z"
      fill="#EB001B"
    />
    <path
      d="M14.03 13.963A8.58 8.58 0 0019.617 16C24.248 16 28 12.418 28 8s-3.753-8-8.382-8a8.58 8.58 0 00-5.588 2.037C15.744 3.502 16.823 5.631 16.823 8c0 2.37-1.08 4.498-2.795 5.963z"
      fill="#F79E1B"
    />
  </svg>
</div>
`;

exports[`PaymentLogo w/ any props 1`] = `
<div
  class="PaymentLogostyles__Box-sc-1op762m-0 dMNKFO className"
  id="alarm"
  title="title"
>
  <svg
    height="16"
    viewBox="0 0 28 16"
    width="28"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      clip-rule="evenodd"
      d="M9.713 1.997h9.313v11.98H9.713V1.996z"
      fill="#FF5F00"
      fill-rule="evenodd"
    />
    <path
      d="M13.97 13.963A8.58 8.58 0 018.383 16C3.752 16 0 12.418 0 8s3.753-8 8.382-8a8.58 8.58 0 015.588 2.037C12.256 3.502 11.176 5.631 11.176 8c0 2.37 1.08 4.498 2.794 5.963z"
      fill="#EB001B"
    />
    <path
      d="M14.03 13.963A8.58 8.58 0 0019.617 16C24.248 16 28 12.418 28 8s-3.753-8-8.382-8a8.58 8.58 0 00-5.588 2.037C15.744 3.502 16.823 5.631 16.823 8c0 2.37-1.08 4.498-2.795 5.963z"
      fill="#F79E1B"
    />
  </svg>
</div>
`;

exports[`PaymentLogo w/ children 1`] = `
<div
  class="PaymentLogostyles__Box-sc-1op762m-0 jmiTAj"
>
  Logo
</div>
`;

exports[`PaymentLogo w/ clickable 1`] = `
<div
  class="PaymentLogostyles__Box-sc-1op762m-0 dMNKFO"
>
  <svg
    height="16"
    viewBox="0 0 28 16"
    width="28"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      clip-rule="evenodd"
      d="M9.713 1.997h9.313v11.98H9.713V1.996z"
      fill="#FF5F00"
      fill-rule="evenodd"
    />
    <path
      d="M13.97 13.963A8.58 8.58 0 018.383 16C3.752 16 0 12.418 0 8s3.753-8 8.382-8a8.58 8.58 0 015.588 2.037C12.256 3.502 11.176 5.631 11.176 8c0 2.37 1.08 4.498 2.794 5.963z"
      fill="#EB001B"
    />
    <path
      d="M14.03 13.963A8.58 8.58 0 0019.617 16C24.248 16 28 12.418 28 8s-3.753-8-8.382-8a8.58 8.58 0 00-5.588 2.037C15.744 3.502 16.823 5.631 16.823 8c0 2.37-1.08 4.498-2.795 5.963z"
      fill="#F79E1B"
    />
  </svg>
</div>
`;

exports[`PaymentLogo w/ design: ghost 1`] = `
<div
  class="PaymentLogostyles__Box-sc-1op762m-0 kQEiiG"
>
  <svg
    height="16"
    viewBox="0 0 28 16"
    width="28"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      clip-rule="evenodd"
      d="M9.713 1.997h9.313v11.98H9.713V1.996z"
      fill="#FF5F00"
      fill-rule="evenodd"
    />
    <path
      d="M13.97 13.963A8.58 8.58 0 018.383 16C3.752 16 0 12.418 0 8s3.753-8 8.382-8a8.58 8.58 0 015.588 2.037C12.256 3.502 11.176 5.631 11.176 8c0 2.37 1.08 4.498 2.794 5.963z"
      fill="#EB001B"
    />
    <path
      d="M14.03 13.963A8.58 8.58 0 0019.617 16C24.248 16 28 12.418 28 8s-3.753-8-8.382-8a8.58 8.58 0 00-5.588 2.037C15.744 3.502 16.823 5.631 16.823 8c0 2.37-1.08 4.498-2.795 5.963z"
      fill="#F79E1B"
    />
  </svg>
</div>
`;

exports[`PaymentLogo w/ design: whitebox 1`] = `
<div
  class="PaymentLogostyles__Box-sc-1op762m-0 gwGHQQ"
>
  <svg
    height="16"
    viewBox="0 0 28 16"
    width="28"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      clip-rule="evenodd"
      d="M9.713 1.997h9.313v11.98H9.713V1.996z"
      fill="#FF5F00"
      fill-rule="evenodd"
    />
    <path
      d="M13.97 13.963A8.58 8.58 0 018.383 16C3.752 16 0 12.418 0 8s3.753-8 8.382-8a8.58 8.58 0 015.588 2.037C12.256 3.502 11.176 5.631 11.176 8c0 2.37 1.08 4.498 2.794 5.963z"
      fill="#EB001B"
    />
    <path
      d="M14.03 13.963A8.58 8.58 0 0019.617 16C24.248 16 28 12.418 28 8s-3.753-8-8.382-8a8.58 8.58 0 00-5.588 2.037C15.744 3.502 16.823 5.631 16.823 8c0 2.37-1.08 4.498-2.795 5.963z"
      fill="#F79E1B"
    />
  </svg>
</div>
`;

exports[`PaymentLogo w/ design: wildsand 1`] = `
<div
  class="PaymentLogostyles__Box-sc-1op762m-0 jmiTAj"
>
  <svg
    height="16"
    viewBox="0 0 28 16"
    width="28"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      clip-rule="evenodd"
      d="M9.713 1.997h9.313v11.98H9.713V1.996z"
      fill="#FF5F00"
      fill-rule="evenodd"
    />
    <path
      d="M13.97 13.963A8.58 8.58 0 018.383 16C3.752 16 0 12.418 0 8s3.753-8 8.382-8a8.58 8.58 0 015.588 2.037C12.256 3.502 11.176 5.631 11.176 8c0 2.37 1.08 4.498 2.794 5.963z"
      fill="#EB001B"
    />
    <path
      d="M14.03 13.963A8.58 8.58 0 0019.617 16C24.248 16 28 12.418 28 8s-3.753-8-8.382-8a8.58 8.58 0 00-5.588 2.037C15.744 3.502 16.823 5.631 16.823 8c0 2.37-1.08 4.498-2.795 5.963z"
      fill="#F79E1B"
    />
  </svg>
</div>
`;

exports[`PaymentLogo w/ logo: mastercard 1`] = `
<div
  class="PaymentLogostyles__Box-sc-1op762m-0 jmiTAj"
>
  <svg
    height="16"
    viewBox="0 0 28 16"
    width="28"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      clip-rule="evenodd"
      d="M9.713 1.997h9.313v11.98H9.713V1.996z"
      fill="#FF5F00"
      fill-rule="evenodd"
    />
    <path
      d="M13.97 13.963A8.58 8.58 0 018.383 16C3.752 16 0 12.418 0 8s3.753-8 8.382-8a8.58 8.58 0 015.588 2.037C12.256 3.502 11.176 5.631 11.176 8c0 2.37 1.08 4.498 2.794 5.963z"
      fill="#EB001B"
    />
    <path
      d="M14.03 13.963A8.58 8.58 0 0019.617 16C24.248 16 28 12.418 28 8s-3.753-8-8.382-8a8.58 8.58 0 00-5.588 2.037C15.744 3.502 16.823 5.631 16.823 8c0 2.37-1.08 4.498-2.795 5.963z"
      fill="#F79E1B"
    />
  </svg>
</div>
`;

exports[`PaymentLogo w/ logo: mir 1`] = `
<div
  class="PaymentLogostyles__Box-sc-1op762m-0 jmiTAj"
>
  <svg
    height="11"
    viewBox="0 0 57 16"
    width="38.5"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      clip-rule="evenodd"
      d="M31.213 1.842l-3.467 7.661h-.385V.097h-5.009v15.806h4.238c1.156 0 2.12-.679 2.6-1.649l3.468-7.66h.385v9.309h5.009V.097h-4.238c-1.156 0-2.12.679-2.6 1.745zM12.528 2.23L10.41 9.503h-.385L7.905 2.23C7.52.97 6.46.194 5.208.194H.2V16h5.008V6.69h.386L8.483 16h3.564l2.89-9.31h.384V16h5.009V.097H15.32c-1.348 0-2.504.873-2.793 2.133zM40.17 7.273v8.63h5.009v-5.042h5.393c2.312 0 4.335-1.455 5.009-3.588H40.17z"
      fill="#24A23F"
      fill-rule="evenodd"
    />
    <path
      clip-rule="evenodd"
      d="M40.556 0c.628 3.44 3.617 6.045 7.214 6.045h8.03c.065-.326.1-.662.1-1.008C55.9 2.256 53.66 0 50.897 0H40.556z"
      fill="#4A90E2"
      fill-rule="evenodd"
    />
  </svg>
</div>
`;

exports[`PaymentLogo w/ logo: visa 1`] = `
<div
  class="PaymentLogostyles__Box-sc-1op762m-0 jmiTAj"
>
  <svg
    height="11"
    viewBox="0 0 57 19"
    width="33.91"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M40.915 1.448l-.654 3.785c-1.963-1.082-6.431-1.338-6.431.654 0 .792 1.01 1.336 2.23 1.992 1.847.995 4.172 2.247 4.172 5.008 0 4.411-4.723 6.033-7.853 6.033-3.131 0-5.18-.995-5.18-.995l.683-3.957c1.878 1.508 7.57 1.935 7.57-.455 0-1.004-1.12-1.596-2.42-2.283-1.792-.948-3.926-2.077-3.926-4.716 0-4.838 5.406-5.834 7.683-5.834 2.106 0 4.126.768 4.126.768zM11.552 13.115L15.925.995h4.937l-7.299 17.621h-4.78L4.798 3.308C2.212 1.753 0 1.293 0 1.293L.084.995h7.23c1.99 0 2.177 1.587 2.177 1.587l1.544 7.924.517 2.609zM19.785 18.616h4.588L27.24.996h-4.588l-2.867 17.62z"
      fill="#20347D"
    />
    <path
      clip-rule="evenodd"
      d="M52.01 18.616h4.213L52.55.996h-3.688c-1.704 0-2.118 1.312-2.118 1.312l-6.842 16.308h4.782L45.64 16h5.832l.538 2.617zM49.373 5.788l-2.411 6.595h3.767l-1.356-6.595z"
      fill="#20347D"
      fill-rule="evenodd"
    />
  </svg>
</div>
`;

exports[`PaymentLogo w/ size: l 1`] = `
<div
  class="PaymentLogostyles__Box-sc-1op762m-0 cYXKph"
>
  <svg
    height="20"
    viewBox="0 0 28 16"
    width="35"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      clip-rule="evenodd"
      d="M9.713 1.997h9.313v11.98H9.713V1.996z"
      fill="#FF5F00"
      fill-rule="evenodd"
    />
    <path
      d="M13.97 13.963A8.58 8.58 0 018.383 16C3.752 16 0 12.418 0 8s3.753-8 8.382-8a8.58 8.58 0 015.588 2.037C12.256 3.502 11.176 5.631 11.176 8c0 2.37 1.08 4.498 2.794 5.963z"
      fill="#EB001B"
    />
    <path
      d="M14.03 13.963A8.58 8.58 0 0019.617 16C24.248 16 28 12.418 28 8s-3.753-8-8.382-8a8.58 8.58 0 00-5.588 2.037C15.744 3.502 16.823 5.631 16.823 8c0 2.37-1.08 4.498-2.795 5.963z"
      fill="#F79E1B"
    />
  </svg>
</div>
`;

exports[`PaymentLogo w/ size: m 1`] = `
<div
  class="PaymentLogostyles__Box-sc-1op762m-0 jmiTAj"
>
  <svg
    height="16"
    viewBox="0 0 28 16"
    width="28"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      clip-rule="evenodd"
      d="M9.713 1.997h9.313v11.98H9.713V1.996z"
      fill="#FF5F00"
      fill-rule="evenodd"
    />
    <path
      d="M13.97 13.963A8.58 8.58 0 018.383 16C3.752 16 0 12.418 0 8s3.753-8 8.382-8a8.58 8.58 0 015.588 2.037C12.256 3.502 11.176 5.631 11.176 8c0 2.37 1.08 4.498 2.794 5.963z"
      fill="#EB001B"
    />
    <path
      d="M14.03 13.963A8.58 8.58 0 0019.617 16C24.248 16 28 12.418 28 8s-3.753-8-8.382-8a8.58 8.58 0 00-5.588 2.037C15.744 3.502 16.823 5.631 16.823 8c0 2.37-1.08 4.498-2.795 5.963z"
      fill="#F79E1B"
    />
  </svg>
</div>
`;

exports[`PaymentLogo w/ size: s 1`] = `
<div
  class="PaymentLogostyles__Box-sc-1op762m-0 loJfsW"
>
  <svg
    height="12"
    viewBox="0 0 28 16"
    width="21"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      clip-rule="evenodd"
      d="M9.713 1.997h9.313v11.98H9.713V1.996z"
      fill="#FF5F00"
      fill-rule="evenodd"
    />
    <path
      d="M13.97 13.963A8.58 8.58 0 018.383 16C3.752 16 0 12.418 0 8s3.753-8 8.382-8a8.58 8.58 0 015.588 2.037C12.256 3.502 11.176 5.631 11.176 8c0 2.37 1.08 4.498 2.794 5.963z"
      fill="#EB001B"
    />
    <path
      d="M14.03 13.963A8.58 8.58 0 0019.617 16C24.248 16 28 12.418 28 8s-3.753-8-8.382-8a8.58 8.58 0 00-5.588 2.037C15.744 3.502 16.823 5.631 16.823 8c0 2.37-1.08 4.498-2.795 5.963z"
      fill="#F79E1B"
    />
  </svg>
</div>
`;

exports[`PaymentLogo w/ size: xl 1`] = `
<div
  class="PaymentLogostyles__Box-sc-1op762m-0 OiIjv"
>
  <svg
    height="24"
    viewBox="0 0 28 16"
    width="42"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      clip-rule="evenodd"
      d="M9.713 1.997h9.313v11.98H9.713V1.996z"
      fill="#FF5F00"
      fill-rule="evenodd"
    />
    <path
      d="M13.97 13.963A8.58 8.58 0 018.383 16C3.752 16 0 12.418 0 8s3.753-8 8.382-8a8.58 8.58 0 015.588 2.037C12.256 3.502 11.176 5.631 11.176 8c0 2.37 1.08 4.498 2.794 5.963z"
      fill="#EB001B"
    />
    <path
      d="M14.03 13.963A8.58 8.58 0 0019.617 16C24.248 16 28 12.418 28 8s-3.753-8-8.382-8a8.58 8.58 0 00-5.588 2.037C15.744 3.502 16.823 5.631 16.823 8c0 2.37-1.08 4.498-2.795 5.963z"
      fill="#F79E1B"
    />
  </svg>
</div>
`;

exports[`PaymentLogo w/ size: xs 1`] = `
<div
  class="PaymentLogostyles__Box-sc-1op762m-0 gzDLys"
>
  <svg
    height="8"
    viewBox="0 0 28 16"
    width="14"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      clip-rule="evenodd"
      d="M9.713 1.997h9.313v11.98H9.713V1.996z"
      fill="#FF5F00"
      fill-rule="evenodd"
    />
    <path
      d="M13.97 13.963A8.58 8.58 0 018.383 16C3.752 16 0 12.418 0 8s3.753-8 8.382-8a8.58 8.58 0 015.588 2.037C12.256 3.502 11.176 5.631 11.176 8c0 2.37 1.08 4.498 2.794 5.963z"
      fill="#EB001B"
    />
    <path
      d="M14.03 13.963A8.58 8.58 0 0019.617 16C24.248 16 28 12.418 28 8s-3.753-8-8.382-8a8.58 8.58 0 00-5.588 2.037C15.744 3.502 16.823 5.631 16.823 8c0 2.37-1.08 4.498-2.795 5.963z"
      fill="#F79E1B"
    />
  </svg>
</div>
`;
